rosbag2_cpp/src/rosbag2_cpp/compression_options.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <string>
+
+#include "rosbag2_cpp/compression_options.hpp"
+#include "rosbag2_cpp/logging.hpp"
+
+namespace rosbag2_cpp
+{
+
+namespace
+{
+
+  constexpr const char kCompressionModeNoneStr[] = "NONE";
+  constexpr const char kCompressionModeFileStr[] = "FILE";
+  constexpr const char kCompressionModeMessageStr[] = "MESSAGE";
+
+  std::string to_upper(const std::string & text)
+  {
+    std::string uppercase_text = text;
+    transform(uppercase_text.begin(), uppercase_text.end(), uppercase_text.begin(), ::toupper);
```

Comment:
Does `transform` exist in the `std` namespace?
If it does, explicitly call from it
```c++
std::transform(...)
```